web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink, mapObjEntries, formatRange } from '../../utils';
+
+interface RulePanelProps {
+  rule: Rule;
+  styles?: CSSProperties;
+  tag?: keyof JSX.IntrinsicElements;
+}
+
+export const RulePanel: FC<RulePanelProps> = ({ rule, styles = {}, tag }) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels } = rule;
+  const style = { background: '#f5f5f5', ...styles };
+  return (
+    <Tag {...(tag ? { style } : {})}>
+      <pre className="m-0" style={tag ? undefined : style}>
+        <code>
+          {rule.type === 'alerting' ? 'alert: ' : 'record: '}
+          <Link
+            to={createExpressionLink(
+              rule.type === 'alerting' ? `ALERTS{alertname="${name.replace(/"/g, '\\"')}"}` : name.replace(/"/g, '\\"')
```

Comment:
The quote escaping in the case of `ALERTS{alertname="..."}` is necessary because the alert name may contain quotes and appears in the context of a label value, which is a string enclosed in quotes. Neither of those two contextual conditions apply in the case of a recording rule, where the recorded metric name cannot contain quotes, nor does it appear as part of a string. So in that case we can just omit the escaping:

```suggestion
              rule.type === 'alerting' ? `ALERTS{alertname="${name.replace(/"/g, '\\"')}"}` : name
```

Reply:
Yeah, make sense, thanks. Just push the suggested change.